My code review suggestions

bet45140 opened the pull request:
**Analysis of the program**
Although this project still doesn't have much functionality and only has the drum button available (which I do not see or hear anything happening when the button is clicked), I am familiar with the purpose of the program. The idea of this project is to build an application in which you can press instrument buttons which will earn you cash tips. After accumulating enough cash, you can buy more instruments in which you can then earn even more cash by pressing the new instrument buttons. You will also be able to hear the instrument sounds as you press the buttons.

**Was the program available in UC Github on time?**
This program was available on UC Github on time. However, when I first performed this code review, the project would not build. It does now, though.

**Is the program documented/commented well enough for you to understand?**
There are some comments throughout the code and most of the work is based on concepts we have learned throughout this course. However, in my opinion, there aren't as many comments as there should be. Some simple comments explaining what some of the functions do as well as what the classes represent may be useful.

**Does the program compile?**
The program does compile without issues now.

**Rationale behind your changes.**
All of the code changes I have made are based on Kotlin coding convention best practices, which should be followed while writing a program in Kotlin.

**Three Technical Concepts Learned**
1. Although we have already learned about them in class, I learned more about the concept of using string templates, which are preferred over string concatenation in Kotlin. I have previous experience with C# and Java programming languages, in which I have always used string concatenation with the + operator. Being able to use string templates make concatenating strings much simpler. The string that I have changed in the program is now a string template as shown: ("$id $name $type").

2. I also learned that you should try to avoid having multi-name package names. However, sometimes it makes sense to have these multi-name package names and if this is the case, you should use camelcase for these names. The package name I changed throughout the project is musicApp, which was originally musicapp.

3. Although it wasn't anything I changed, nor was it something I was able to see working (since I can't get the project to build), I was curious about the Kotlin Toast message, shown in the program as: Toast.makeText(baseContext, msg, Toast.LENGTH_SHORT).show(). I searched for this online and learned that a Toast message is a small message that appears on a screen and disappears after a short time.

bet45140 committed:
My code review suggestions

bet45140 commented on app/src/androidTest/java/mobile/programming/musicApp/ExampleInstrumentedTest.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp
+package mobile.programming.musicApp
```
**Technical Suggestion 1:** While creating packages, Kotlin naming conventions states that you should try to avoid multi-name package names, such as musicApp. However, if you do use a multi-name package name such as this one, you should use camelcase, which I have changed throughout the project. This information is found in the following link under the "Naming rules" segment: https://kotlinlang.org/docs/reference/coding-conventions.html#documentation-comments

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/androidTest/java/mobile/programming/musicApp/ExampleInstrumentedTest.kt:
```diff
@@ -19,6 +19,6 @@ class ExampleInstrumentedTest {
         // Context of the app under test.
         val appContext = InstrumentationRegistry.getInstrumentation().targetContext
-        assertEquals("com.example.musicapp", appContext.packageName)
+        assertEquals("com.example.musicApp", appContext.packageName)
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="mobile.programming.musicapp">
+    package="mobile.programming.musicApp">
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -9,7 +9,7 @@
         android:supportsRtl="true"
         android:theme="@style/AppTheme">
-        <activity android:name="mobile.programming.musicapp.MainActivity">
+        <activity android:name="mobile.programming.musicApp.MainActivity">
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/MainActivity.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp
+package mobile.programming.musicApp
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/RetrofitClientInstance.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp
+package mobile.programming.musicApp
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/dao/MusicAppDAO.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp.dao
+package mobile.programming.musicApp.dao
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/dto/Instrument.kt:
```diff
@@ -1,11 +1,11 @@
-package mobile.programming.musicapp.dto
+package mobile.programming.musicApp.dto
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/dto/Instrument.kt:
```diff
@@ -1,11 +1,11 @@
     override fun toString(): String{
 
-        return (id.toString() + " " + name + " " + type)
+        return ("$id $name $type")
```
**Technical Suggestion 2:** According to Kotlin coding conventions, the preferred method of combining strings like this is to use string templates (which is what I have changed to in the code) as opposed to string concatenation (which is what you are using). This information can be found here under the "Using strings" segment: https://kotlinlang.org/docs/reference/coding-conventions.html#documentation-comments

bet45140 commented on app/src/main/java/mobile/programming/musicApp/service/InstrumentService.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp.service
+package mobile.programming.musicApp.service
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/main/java/mobile/programming/musicApp/ui/main/MainViewModel.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp.ui.main
+package mobile.programming.musicApp.ui.main
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/test/java/mobile/programming/musicApp/InstrumentDataIntegrationTest.kt:
```diff
@@ -1,4 +1,4 @@
-package mobile.programming.musicapp
+package mobile.programming.musicApp
```
Here, I have also changed the multi-name package name to camelcase.

stoberdl replied:
Good noting, I did not think about correct naming conventions when renaming the project for Firebase

bet45140 commented on app/src/test/java/mobile/programming/musicApp/InstrumentDataIntegrationTest.kt:
```diff
@@ -32,10 +32,10 @@ class InstrumentDataIntegrationTest {
 
     private fun checkIfInstrumentDataIsEmpty() {
-        var instrumentlist = ArrayList<Instrument>()
+        var instrumentList = ArrayList<Instrument>()
```
**Technical Suggestion 3:** According to Kotlin coding conventions, variables and parameters should always be written in lower camelcase. This was probably a simple mistake, but this convention should always be followed for readability purposes. This coding convention can be found under the Variables and Parameters section at: https://github.com/raywenderlich/kotlin-style-guide#variables--parameters

bet45140 commented on app/src/test/java/mobile/programming/musicApp/InstrumentDataIntegrationTest.kt:
```diff
@@ -32,10 +32,10 @@ class InstrumentDataIntegrationTest {
         mvm.instruments.observeForever {
 
-            instrumentlist = it
+            instrumentList = it
```
This is just another instance of the instrumentList variable that needs to be changed to lower camelcase.

stoberdl commented:
Sorry about that. It seems my other group member committed the notification code without checking if it built. I have fixed his code and it should now run correctly with notification functionality.

bet45140 committed:
Update MainActivity.kt

Since this file was causing a conflict with the merge, I copied what you had in the Master branch version of this file into this file.
